From 0c03613fcebd3e7256a4cd3c7cffadc1a7f5932f Mon Sep 17 00:00:00 2001
From: Wenchao Hao <haowenchao@huawei.com>
Date: Sat, 29 Jan 2022 00:22:55 +0800
Subject: [PATCH] kabi:fuse: reserve space for future expansion
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0c03613fcebd3e7256a4cd3c7cffadc1a7f5932f
Modified-by-SEL: No


euleros inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4PF3R?from=project-issue

-------------------------------------------------

Reserve space for future fuse kabi expansion

Signed-off-by: Wenchao Hao <haowenchao@huawei.com>
Reviewed-by: Wu Bo <wubo40@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/fuse/fuse_i.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/fs/fuse/fuse_i.h b/fs/fuse/fuse_i.h
index b159d8b5e893..ed71ef6fbc9d 100644
--- a/fs/fuse/fuse_i.h
+++ b/fs/fuse/fuse_i.h
@@ -31,6 +31,7 @@
 #include <linux/pid_namespace.h>
 #include <linux/refcount.h>
 #include <linux/user_namespace.h>
+#include <linux/kabi.h>
 
 /** Default max number of pages that can be used in a single read request */
 #define FUSE_DEFAULT_MAX_PAGES_PER_REQ 32
@@ -162,6 +163,8 @@ struct fuse_inode {
 	 */
 	struct fuse_inode_dax *dax;
 #endif
+
+	KABI_RESERVE(1)
 };
 
 /** FUSE inode state bits */
@@ -485,6 +488,8 @@ struct fuse_dev {
 
 	/** list entry on fc->devices */
 	struct list_head entry;
+
+	KABI_RESERVE(1)
 };
 
 struct fuse_fs_context {
@@ -790,6 +795,11 @@ struct fuse_conn {
 
 	/** List of filesystems using this connection */
 	struct list_head mounts;
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 /*
-- 
2.26.2

